Automattic/wp-calypso
# Promotions: Try a data-driven approach

**NOTE: This PR supersedes #18851 and should be reviewed instead of it.**

This tries out a data-driven approach to promotions UI since so much of
the UI is different based on the promotion type selected.

This will also make adding the many different constraints in much easier and clearer in the component code.

To Test:
1. `npm test`
2. `npm start`
3. Visit `http://calypso.localhost:3000/store/promotion/<site url>`
4. Change the promotion type selection and observe the fields below changing.
5. Attempt to enter text into the fields below.
6. Attempt to enter an invalid percent.
7. Attempt to enter a negative currency amount.

![new_promotion_ _coderkevin_ _wordpress_com](https://user-images.githubusercontent.com/945228/31925033-5ffa47f8-b84b-11e7-80e6-531c32f492bf.png)

![new_promotion_ _coderkevin_ _wordpress_com](https://user-images.githubusercontent.com/945228/31925046-840b24d2-b84b-11e7-94fc-fcbb1f8c65d4.png)

![new_promotion_ _coderkevin_ _wordpress_com](https://user-images.githubusercontent.com/945228/31925051-92fbdfc2-b84b-11e7-827e-b0acdc3347a9.png)
